### PR TITLE
implement TestValidateCAPIAuthTokenAccessHandler

### PR DESCRIPTION
**Pull request by eaudetcobello**
* TestValidateCAPIAuthTokenAccessHandler
* rename WithDB to WithState and move to a new location with package name microcluster_testenv
* WithState gives access to the whole state.State type instead of state.State.Database() only

Depends on https://github.com/canonical/k8s-snap/pull/880

### CODE REVIEW

File: src/k8s/pkg/k8sd/api/capi_access_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+import (
+	"context"
+	"database/sql"
+	"net/http"
+	"testing"
+
+	"github.com/canonical/k8s/pkg/k8sd/api"
+	"github.com/canonical/k8s/pkg/k8sd/database"
+	microcluster_testenv "github.com/canonical/k8s/pkg/utils/microcluster"
```
Comment (bschimke95) [inline src/k8s/pkg/k8sd/api/capi_access_handler_test.go:11]: nit: not sure if we need to be that verbose but fine for me

**Review by bschimke95 — APPROVED** [review]
LGTM

File: src/k8s/pkg/k8sd/api/capi_access_handler_test.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+					err = s.Database().Transaction(ctx, func(ctx context.Context, tx *sql.Tx) error {
+						return database.SetClusterAPIToken(ctx, tx, tc.tokenDBContent)
+					})
+					g.Expect(err).To(Not(HaveOccurred()))
+				}
+
+				req := &http.Request{
+					Header: make(http.Header),
+				}
+				req.Header.Set(api.CAPITokenHeaderName, tc.tokenHeaderContent)
```
Comment (HomayoonAlimohammadi) [inline src/k8s/pkg/k8sd/api/capi_access_handler_test.go:75]: Ah I really wish we could have the `node-token` in the k8s API as well...
Would you mind adding it or making a card to add it?

**Review by HomayoonAlimohammadi — APPROVED** [review]
Thanks a lot @eaudetcobello, LGTM!

File: src/k8s/pkg/k8sd/api/capi_access_handler_test.go (hunk 1 above)
Comment (eaudetcobello) [reply src/k8s/pkg/k8sd/api/capi_access_handler_test.go]: Changed to testenv

**Pull request merged**